src/sgame/sg_bot_nav.cpp

```diff
@@ -586,9 +586,17 @@ static bool BotShouldJump( gentity_t *self, const gentity_t *blocker, const glm:
 	//check if jumping will clear us of entity
 	trap_Trace( &trace, origin, playerMins, playerMaxs, end, self->s.number, MASK_SHOT, 0 );
 
+	classAttributes_t const* pcl = BG_Class( pClass );
+	bool ladder = ( pcl->abilities & SCA_CANUSELADDERS )
```

Comment:
The trace here goes forward 20 units but PM_CheckLadder only looks forward 1 unit. Maybe there should be a condition on the trace distance being small enough?

Reply:
I'm afraid that using a shorter trace would make the bot trigger avoidance maneuvers, and thus would prevent it to climb at ladder.

Reply:
I'm not saying to use a shorter trace. Rather to check how far the hit point is using trace fraction or whatever

Reply:
I don't get you... why would that be useful? Is there is a known problem with this PR?

Reply:
What stops the bots from uselessly jumping and wasting stamina when they aren't close enough to climb?

Reply:
have you seen this with this PR merged? Condisering unv's slippy physics, there are very, very little chances for this to happen.
The code is certainly nor perfect, but I can't do perfect handling with the current status, because code handling AI moves is just a pile of hacks.
I will need further refactoring to get wallwalking done, and it won't be a 5-line patch, it will take weeks of discussions and talking to be merged. I know it in advance.
In the meantime, I need this merged because this will allow to check for regressions, and because in the meantime it will allow the bots to use ladders.

In any case, this requires off-mesh connections to be placed, so no, there is no "loss" of stamina to expect.

Reply:
All things considered I think we could probably merge this if there's no technical reason other than "bots might waste stamina" - this will require testing and I am willing to merge it into bunker once it's upstream (ping @sweet235 for his tactic server too).

If worst comes to the worst, we can revert this, but it looks like @bmorel is laying the groundwork for more improvements.

It certainly cannot break things more than the revamped navmesh generator did in 0.54.0 ...